### DISCUSSION TITLE
EMQX Auth/ ACL Auth API timeouts

### DISCUSSION THREAD

**Discussion by YashJainSC** [emqx/emqx#8565] [Q&A] [answered]
We are noticing intermittent spikes in timeout errors

logs:

`ClientIp@ip:ports [Auth http] Request auth path: /service/v1/emqx/auth, error: timeout`

Also noticing this logs whenever timeouts error occure

`ehttpc: Received 'gun_error' message from unknown stream ref`

I am not sure whether this is due to connection timeout or request timeout, how to figure that out ?

Anything we can do here, currently we are using default settings, will increasing pool size help ? currently for auth + acl auth APIs we have about ~4000 requests per second

**Comment by lafirest** [comment, marked as answer]
This is because your web server response is too slow, you can increase timeout, or deploy LB in front of your web server

**Reply by YashJainSC** [reply]
We have LB in place, and you are right service giving timeouts, we have identified that and working on fixing it.

just wanted to know what this error exactly means 

`ehttpc: Received 'gun_error' message from unknown stream ref`

**Reply by zmstone** [reply]
this error is harmless.
it happens when a stale http error comes back for a canceled http request.
ehttpc only waits certain amount of time (the timeout) for each sent request, once the request is timed out, it tries to send a cancellation to the underlying http library to 'cancel' the request.
this cancellation however is async, so there is still a chance for http responses (error response in this case) to reach ehttpc.

we'll consider changing this logging level to `debug` or `info`

**Reply by YashJainSC** [reply]
okay thanks got it
